### PR TITLE
sqlddl refactor

### PR DESCRIPTION
**Pull request by cmungall**

### CODE REVIEW

File: linkml/generators/sqltablegen.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        self.rename_foreign_keys = rename_foreign_keys
+        self.direct_mapping = direct_mapping
+
+    def generate_ddl(self, naming_policy: SqlNamingPolicy = None, **kwargs) -> str:
+        ddl_str = ""
+        def dump(sql, *multiparams, **params):
+            nonlocal ddl_str
+            ddl_str += f"{str(sql.compile(dialect=engine.dialect)).rstrip()};"
+
+        engine = create_mock_engine(
```
Comment (sujaypatil96) [inline linkml/generators/sqltablegen.py:142]: Missing import statement.

```python
from sqlalchemy import create_mock_engine
```

**Review by sujaypatil96 — COMMENTED** [review]
Review process still in progress. Follow PR #500.

File: linkml/generators/sqltablegen.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                        args = [ForeignKey(fk)]
+                    field_type = self.get_sql_range(s, schema)
+                    col = Column(sql_name(sn), field_type, *args, primary_key=is_pk, nullable=not s.required)
+                    cols.append(col)
+                Table(sql_name(cn), schema_metadata, *cols)
+        schema_metadata.create_all(engine)
+        return ddl_str
+
+    # TODO: merge with code from sqlddlgen
+    def get_sql_range(self, slot: SlotDefinition, schema: SchemaDefinition):
```
Comment (sujaypatil96) [inline linkml/generators/sqltablegen.py:187]: Can we make `schema: SchemaDefinition = None`?

And add an if block in the body like such:

```
if not schema:
    schema = self.schema
```

Comment (cmungall) [reply linkml/generators/sqltablegen.py:187]: yes

**Pull request merged**